core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -85,8 +85,16 @@ public void run(final ActionContext context) throws Exception {
 
       LOG.debug("Output: {}", out);
       LOG.error("Errors: {}", err);
+      Integer exitCode = session.getExitStatus();
+      if (exitCode != null && exitCode > 0) {
```

Comment:
most systems have non-negative exit codes, but its still safer just to check the exitCode != 0 in case negative exit codes are allowed.